### PR TITLE
fix: reset error boundary after navigation

### LINKED ISSUE #1539: Go Home button at error screen is not working
## Bug description
If by any case the user ends up at the boundary screen the "Go Home" button doesn't work so it's stuck there

## Environment
 - Browser: all
 - Wallet: all
 - Chain: all

## Steps to reproduce
 1. Go to https://app.safe.global/apps?safe=gor:0x4127839cdf4F73d9fC9a2C2861d8d1799e9DF40C&appUrl=appUrl%3Djavascript%3AstartHack%28%29
 2. Try to get out of the error screen by clicking Go Home
 3. No navigation is triggered

## Expected result
The user is able to go to the Home screen

## Obtained result
Nothing happens

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #1539

## How this PR fixes it

A standard link is used instead of an internally routed one.

## How to test it

Navigate to `/apps?safe=gor:0x4127839cdf4F73d9fC9a2C2861d8d1799e9DF40C&appUrl=appUrl%3Djavascript%3AstartHack%28%29)` and click "Go Home". Observe that the `/welcome` page is opened and error stack is reset.

### CODE REVIEW

File: src/services/pairing/QRModal.tsx [hunk 1]
```diff
@@ -59,7 +60,7 @@ const Modal = ({ uri, cb }: { uri: string; cb: () => void }) => {
 
   return (
     <StoreHydrator>
-      <AppProviders>
+      <ThemeProvider theme={theme}>
```
Comment (iamacook) [inline src/services/pairing/QRModal.tsx:63]: The `router` object is not accessible here.

File: src/components/common/ErrorBoundary/index.tsx [hunk 2]
```diff
@@ -37,8 +41,18 @@ const ErrorBoundary: FallbackRender = ({ error, componentStack }) => {
...
-          <Link href={AppRoutes.welcome} passHref target="_blank" rel="noopener noreferrer" color="primary">
-            <MuiLink>Go Home</MuiLink>
+          <Link href={AppRoutes.welcome} passHref color="primary">
+            <MuiLink
+              onClick={(e) => {
+                e.stopPropagation()
+
+                // We need to wait for navigation to finish otherwise error will be thrown
+                // from the current page again
+                router.push(AppRoutes.welcome).then(resetError)
```
Comment (katspaugh) [inline src/components/common/ErrorBoundary/index.tsx:51]: Why does it have to be a JS function and not just a link?
I would even make it an actual link, that does a full page reload.

File: src/components/common/ErrorBoundary/index.tsx [hunk 3]
```diff
@@ -37,8 +41,18 @@ const ErrorBoundary: FallbackRender = ({ error, componentStack }) => {
...
+              }}
+            >
+              Go Home
```
Comment (katspaugh) [inline src/components/common/ErrorBoundary/index.tsx:54]:
```suggestion
              Go home
```

File: src/components/common/ErrorBoundary/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/common/ErrorBoundary/index.tsx:51]: Resetting the error occurs before the navigation and the error is thrown again from what was the previous page.

If you'd sooner use a link then this is unnecessary then. I'll change it.

**Review by katspaugh — APPROVED** [review]
👍

**Review by usame-algan — APPROVED** [review]
🚀

**Comment by francovenica** [comment]
LGTM
Tried with the URL daniel provided, and used the "Go home" button from there
https://fix_error_boundary_link--webcore.review-web-core.5afe.dev/apps?safe=gor:0x4127839cdf4F73d9fC9a2C2861d8d1799e9DF40C&appUrl=appUrl%3Djavascript%3AstartHack%28%29

**Pull request merged**